utils.go

```diff
@@ -405,11 +550,40 @@ func SanitizeUprobeAddresses(f *elf.File, syms []elf.Symbol) {
 
 // FindSymbolOffsets - Parses the provided file and returns the offsets of the symbols that match the provided pattern
 func FindSymbolOffsets(path string, pattern *regexp.Regexp) ([]elf.Symbol, error) {
+	var syms []elf.Symbol
 	f, syms, err := OpenAndListSymbols(path)
 	if err != nil {
+		libRegex, err := regexp.Compile(path)
+		if err != nil {
+			return nil, err
+		}
+		libs := so.Find(libRegex)
+		if len(libs) == 0 {
+			return nil, fmt.Errorf("Can't find running process that have this lib %s loaded", path)
+		}
+		found := false
+		for _, lib := range libs {
+			for _, pidPath := range lib.PidsPath {
+				pidStr := filepath.Base(pidPath)
+				var pid int
+				pid, err = strconv.Atoi(pidStr)
+				if err != nil {
+					continue
+				}
+				f, syms, err = OpenAndListSymbolsFromPID(pid, path)
+				if err == nil {
+					found = true
+					goto found_syms
+				}
+			}
+		}
+		if !found {
```

Comment:
`found` will always be `false` here, because as soon as it is set to `true`, you use a `goto`.